libs/common-utils/src/fractionUtils.ts

```diff
@@ -169,4 +174,43 @@ export class FractionUtils {
 
     return amount.equalTo(0) ? CurrencyAmount.fromRawAmount(amount.currency, 1) : amount
   }
+
+  static simplify(fraction: Fraction): Fraction {
+    return reduce(trimZeros(fraction))
+  }
+}
+
+/**
+ * Use GCD to reduce the fraction to the smallest possible
+ *
+ * From https://stackoverflow.com/a/65927538
+ *
+ * @param fraction
+ */
+function reduce(fraction: Fraction): Fraction {
+  let a = fraction.numerator
+  let b = fraction.denominator
+  let c: JSBI
+  while (JSBI.notEqual(b, JSBI.BigInt(0))) {
```

Comment:
It's better to cache `JSBI.BigInt(0)` into a const

Reply:
40494100dfb6